static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js

```diff
@@ -38,40 +44,60 @@ const setPageTargeting = (): void => {
     });
 };
 
-const forceRemoveAdSlots = (): Promise<void> => closeDisabledSlots(true);
+// This is specifically a separate function to close-disabled-slots. One is for
+// closing hidden/disabled slots, the other is for graceful recovery when prepare-googletag
+// encounters an error. Here, slots are closed unconditionally.
+const removeAdSlots = (): Promise<void> => {
+    // Get all ad slots
+    const adSlots: Array<Element> = qwery(dfpEnv.adSlotSelector);
 
-const removeAdSlots = (): Promise<void> => closeDisabledSlots(false);
+    return fastdom.write(() =>
+        adSlots.forEach((adSlot: Element) => adSlot.remove())
+    );
+};
+
+const setPublisherProvidedId = (): void => {
+    const user: ?Object = identity.getUserFromCookie();
+    if (user) {
+        const hashedId = sha1.hash(user.id);
+        window.googletag.pubads().setPublisherProvidedId(hashedId);
+    }
+};
 
 const init = (start: () => void, stop: () => void): Promise<void> => {
     const setupAdvertising = (): Promise<void> => {
         addTag(dfpEnv.sonobiEnabled ? 'sonobi' : 'waterfall');
 
+        start();
+
+        // note: fillAdvertSlots isn't synchronous like most buffered cmds, it's a promise. It's put in here to ensure
+        // it strictly follows preceding prepare-googletag work (and the module itself ensures dependencies are
+        // fulfilled), but don't assume fillAdvertSlots is complete when queueing subsequent work using cmd.push
         window.googletag.cmd.push(
-            start,
             setDfpListeners,
             setPageTargeting,
-            stop
+            setPublisherProvidedId,
+            refreshOnResize,
+            () => {
+                fillAdvertSlots().then(stop);
+            }
         );
 
         // Just load googletag. Sonobi's wrapper will already be loaded, and googletag is already added to the window by sonobi.
         return loadScript(config.libs.googletag, { async: false });
     };
 
     if (commercialFeatures.dfpAdvertising) {
```

Comment:
This will be true now, even if ad-free is enabled because it is driven just by `switches.commercial && externalAdvertising && !sensitiveContent`